File: gems/decomplex/examples/php/path-condition.php
<?php
function one($x, $y, $z) {
  if ($x->p() && $y->q() && $z->r()) { go($x); }
}

function two($x, $y, $z) {
  if ($x->p() && $y->q() && $z->r()) { go($x); }
}

function three($x, $y, $z) {
  if ($x->p() && $y->q() && $z->r()) { go($x); }
}

function bug($x, $y, $z) {
  if ($x->p() && $y->q()) { go($x); }
}

File: gems/decomplex/examples/zig/path-condition.zig
pub fn one(x: X, y: Y, z: Z) void {
    if (x.p() and y.q() and z.r()) { go(x); }
}

pub fn two(x: X, y: Y, z: Z) void {
    if (x.p() and y.q() and z.r()) { go(x); }
}

pub fn three(x: X, y: Y, z: Z) void {
    if (x.p() and y.q() and z.r()) { go(x); }
}

pub fn bug(x: X, y: Y, z: Z) void {
    if (x.p() and y.q()) { go(x); }
}
